app/forms/catalog_record_id_form.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "reform/form/coercion"
+
+class CatalogRecordIdForm < Reform::Form
+  ### classes that define a virtual catalog record ID model object and data structure, used in form editing...persistence is in the cocina model
+  class Row
+    attr_accessor :value, :refresh, :catalog, :_destroy
+
+    def initialize(attrs = {})
+      @value = attrs[:value]
+      @refresh = attrs[:refresh]
+      @catalog = attrs[:catalog]
+    end
+
+    def persisted?
+      false
+    end
+  end
+  ###
+
+  feature Reform::Form::Coercion
+
+  collection :catalog_record_ids, populate_if_empty: Row, save: false, virtual: true,
+    prepopulator: ->(*) { catalog_record_ids << CatalogRecordIdForm::Row.new(value: "", refresh: true) if catalog_record_ids.size.zero? } do
+    property :value
+    property :refresh, type: Dry::Types["params.nil"] | Dry::Types["params.bool"]
+    property :_destroy
+  end
+
+  collection :other_catalog_record_ids, save: false, virtual: true do
+    property :value
+    property :catalog
+    property :refresh, type: Dry::Types["params.nil"] | Dry::Types["params.bool"]
+    property :_destroy
+  end
+
+  validate :single_catalog_record_id_refresh
+  validate :unique_catalog_record_id_value
+  validate :valid_catalog_record_id_value
+
+  # needed because our model in this form is a cocina object and not an active record model, and Reform calls `persisted?`
+  def persisted?
+    false
+  end
+
+  def setup_properties!(_options)
+    object_catalog_record_ids = model.identification.catalogLinks.filter_map { |catalog_link| catalog_link if catalog_link.catalog == CatalogRecordId.value }
+
+    self.catalog_record_ids = object_catalog_record_ids.map { |catalog_record_id| CatalogRecordIdForm::Row.new(value: catalog_record_id.catalogRecordId, refresh: catalog_record_id.refresh) }
+
+    other_object_catalog_record_ids = model.identification.catalogLinks.filter_map { |catalog_link| catalog_link if catalog_link.catalog != CatalogRecordId.value }
+
+    self.other_catalog_record_ids = other_object_catalog_record_ids.map { |catalog_record_id| CatalogRecordIdForm::Row.new(value: catalog_record_id.catalogRecordId, refresh: catalog_record_id.refresh, catalog: catalog_record_id.catalog) }
```

Comment:
Added this to help persist Folio catalog links when in Symphony mode, and vice versa.